appcues/src/main/java/com/appcues/Appcues.kt

```diff
@@ -132,14 +133,22 @@ class Appcues internal constructor(koinScope: Scope) {
      * Generate a unique Id for the current user when there is not a known identity to use in
      * the {@link identity(String, Map<String, Any>) identity} call. This will cause the SDK
      * to begin tracking activity and checking for qualified content.
-     *
-     * @param properties Optional properties that provide additional context about the anonymous user.
      */
-    fun anonymous(properties: Map<String, Any>? = null) {
+    fun anonymous() {
         // use the device ID as the default anonymous user ID, unless an override for generating
         // anonymous user IDs is supplied in the config builder
         val anonymousId = config.anonymousIdFactory?.invoke() ?: storage.deviceId
-        identify(true, anonymousId, properties)
+        identify(true, anonymousId, null)
+    }
+
+    /**
+     * This function has been removed. Calling the anonymous function with a properties parameter
+     * is no longer supported. A call to `anonymous()` with no parameters should be used instead.
+     */
+    @Deprecated("properties are no longer supported for anonymous users.", level = ERROR)
+    @Suppress("UnusedPrivateMember")
+    fun anonymous(properties: Map<String, Any>?) {
+        // removed
```

Comment:
@andretortolano does this match your understanding for how to give a deprecation error with more info to any existing callers? Did [something similar](https://github.com/appcues/appcues-ios-sdk/pull/349/files#diff-864ffdf7e8b525ca234734f7b71d722506418595a7c147f2597998eadee3a246) on iOS side for this change.

Reply:
I guess we could keep the level as warning since it wont break any code if they still use this one. you could make so there is a link to the correct function with ReplaceWith

@Deprecated("properties are no longer supported for anonymous users.", replaceWith = ReplaceWith("anonymous()"))

https://dev.to/mreichelt/the-hidden-kotlin-gem-you-didn-t-think-you-ll-love-deprecations-with-replacewith-3blo

Reply:
it seems like with that, Android Studio will suggest the change

Reply:
Yeah, I think the idea is that we are explicitly removing functionality here, so the more forceful error is appropriate, versus a deprecation warning or internally dropping the args, which might lead to more subtle unexpected behavior where it is still "allowed" but will stop functioning as it was expected before.  @mmaatttt any thoughts?

Reply:
Yeah it should be an compile error not a warning